### ISSUE TITLE
Source license?

### ISSUE THREAD

**Issue by poVoq** [gloomyson/StarCraft#3]
What license would be this code under?

Would there anything be in it that would prevent a fully FOSS version if one would make a separate graphics and sounds set?

**Comment by gloomyson** [comment]
License is not added yet, and I don't mind if someone use my code to build a new RTS, but he needs to mention that he's using my code.

**Comment by anko** [comment]
http://choosealicense.com/ — [MIT/X11](https://opensource.org/licenses/MIT) or [ISC](https://opensource.org/licenses/ISC) perhaps?

Does @Blizzard mind?

**Comment by roblabla** [comment]
Well the source code's license isn't something Blizzard would have a say in, since it's not theirs (the assets are a different story though).

If you want attribution, MIT or BSD are very liberal licenses that work great.

**Comment by gloomyson** [comment]
I admit the sprite images and sounds are extracted from original StarCraft dat files, if Blizzard warn me by email for some copyright issues I have to remove all the img and bgm from this project; but the js source code of this game engine are completely owned by me, I write these code from scratch, nothing to do with Blizzard.

**Comment by colindean** [comment]
Depending on how you extracted the sprites and audio, you might want to include in the repo instructions how to do that in case someone wants to do it themselves.

**Comment by gloomyson** [comment]
I don't think reveal the starcraft data parse method is a good idea, Blizzard will also be unhappy if their sprites extraction method revealed to public, besides these sprites sources, the whole game has nothing to do with Blizzard, I created it from scratch, I can get an artist to create another sprite suites then this game will become a new RTS, totally belongs to me.

**Comment by cigraphics** [comment]
You are a GOD!! This is amazing and it brings back lots of memories! Thanks!

**Comment by starthal** [comment]
>Blizzard will also be unhappy if their sprites extraction method revealed to public

Doesn't matter - code to extract the sprites from the `dat` files does not violate any copyright. No matter how unhappy Blizzard may be there isn't much they could do.

Including the sprites and music, on the other hand, does violate copyright. Whether you post code to extract or not, you should make a clean init of the repo with no assets.

**Comment by roblabla** [comment]
I just want to echo what @starthal said and further say that I believe Blizzard would much prefer you just publish the extract code, rather than publish the assets directly. Blizzard wants it to be complicated for people to get their hands on the assets, and it's much harder if people have to use an extraction script on the executable than it is to clone your repo and copy the image folder :smile_cat:

**Comment by starthal** [comment]
See the help at https://help.github.com/articles/remove-sensitive-data/ if you want to remove the assets without eliminating existing history.

**Comment by gloomyson** [comment]
@starthal You should look this http://www.samods.org/starcraft/modding_tools if you're intrested, but these 3rd party tools can only extract separated sprites, I joined them together, game units cannot show normally if sprites are not in correct position in png

**Comment by starthal** [comment]
My point was that a [DMCA takedown](https://help.github.com/articles/dmca-takedown-policy/) is very likely if this repo gets any more popular, because in its current form it is violating copyright, and it will be unless the sounds and sprites are purged. You may lose whatever GitHub history (issues, PRs, etc.) has accumulated by that point. Anyone forking this runs the same risk.

I advise you to remove the assets from the repo as soon as possible.

**Comment by willsr** [comment]
Please take action soon. Blizzard are pretty serious when it comes to DMCA takedowns. 

https://github.com/github/dmca/blob/master/2013-05-10-Blizzard-Entertainment.md
https://www.eff.org/cases/blizzard-v-bnetd
https://en.wikipedia.org/wiki/MDY_Industries,_LLC_v._Blizzard_Entertainment,_Inc.
https://www.chillingeffects.org/notices/search?page=2&sender_name_facet=Entertainment+Software+Association&term-require-all=true&term=blizzard&utf8=%E2%9C%93

**Comment by gloomyson** [comment]
If remove these assets, how about those forked repos? Will these assets removed too?

**Comment by Far0s** [comment]
Actually, one of your links @willsr speaks of "reverse-engineering" as one of the reasons to take down an a piece of open source software. Doesn't that mean that even if @gloomyson wrote the code himself, the fact that it emulates Blizzard software is enough for them ? I'm not an expert in this domain, I just feel like it could be a problem :/ 

This is really a nice achievement, but yeah, Blizzard seems to suspend those projects with the same enthusiasm as the fans who create thoses sorts of projects.

**Comment by roblabla** [comment]
@Far0s IANAL, but I believe reverse-engineering is legal and protected by law if it is done under fair use, at least in the US.

**Comment by Suvitruf** [comment]
@roblabla nop. DMCA.Especially in US.

**Comment by roblabla** [comment]
@Suvitruf https://www.eff.org/issues/coders/reverse-engineering-faq

Looks like reverse-engineering is legal under fair used unless you gave up your rights to it through an EULA. I don't know much about SC1's EULA to know if it contains such a condition.

**Comment by willsr** [comment]
I have found the StarCraft 1 EULA ![StarCraft 1 EULA.](https://cloud.githubusercontent.com/assets/1422761/9736635/8c0a50dc-5681-11e5-919d-c389e47735a2.jpg) 

It seems this does contain a reverse-engineering clause. IANAL, but, I don't think Blizzard would issue a takedown notice over a guide on how to extract the assets from a legitimate copy of the game. However, they are very likely to bring a takedown over the copied assets.

**Comment by xboi209** [comment]
Isn't there a clause to allow the user to end the agreement(EULA)? If so, @gloomyson could end the agreement and then we tell him what to do?

**Comment by starthal** [comment]
Even given the EULA, I doubt that a guide or a script to extract the assets would be considered reverse engineering in itself. The user *running* the script might be violating the EULA but hosting the script itself should be safe. IANAL

**Comment by ocdtrekkie** [comment]
I definitely think the extraction instructions to pull assets will be fine (for many games, this info is public, and the instructions for StarCraft assets are already likely public at least somewhere).

But the art and stuff, is going to be DMCA'd for sure. Just a matter of time. Seen another super cool project get DMCA'd for using just like the CSS from a proprietary site. Inevitably, someone shows up on the repo who feels it's their job in life to rat people out, and will directly email Blizzard legal about this. :/

I totally just downloaded this though, this is awesome. Good work.

**Comment by ocdtrekkie** [comment]
It's gotten mainstream attention: http://thenextweb.com/apps/2015/09/07/starcraft-now-works-in-the-browser-and-its-amazing/

You'll probably get a DMCA by the end of the week, knowing Blizzard. :/

**Comment by xboi209** [comment]
I have some doubt about how soon Blizzard will send DMCA notices. Yes this is infringing on copyright but Blizzard has mostly abandoned their old games.

**Comment by ocdtrekkie** [comment]
@xboi209 The original StarCraft War Chest is still sold on Blizzard's website today, and can be connected to modern Battle.net accounts.

**Comment by xboi209** [comment]
@ocdtrekkie Yes I'm aware that the game is still being officially sold, but the fact that large private servers have been running for **years** makes me doubt that someone using their artwork and sounds would tick them off.

**Comment by starthal** [comment]
Does a private server infringe upon copyright? I'm not sure how it would.

**Comment by xboi209** [comment]
@starthal They're declared illegal on grounds of piracy for over a decade now

**Comment by Sleaker** [comment]
@xboi209 - the last project that emulated Starcraft/warcraft was told to change their name and remove all of the assets, they weren't shutdown however. But Blizzard forced them to rebrand as Wargus and Stargus.  Look it up.

**Comment by Suvitruf** [comment]
@Sleaker at that time...StarCraft Universe looks fine )

**Comment by xboi209** [comment]
@Sleaker My point is that it's 2015 now and Blizzard has mostly abandoned their old games, it would be a surprise to me if they sent a DMCA notice at all

**Comment by toobulkeh** [comment]
![This gunna be good]
(http://media.giphy.com/media/LFlT04CTtrwc/giphy.gif)

**Comment by Alzarath** [comment]
I'd recommend making a utility to export and compile all of the desired sprites from an installation of StarCraft. This should be do-able with libmpq, I think?

I've been looking forward to a reimplementation of the StarCraft engine for a while now and no one seemed to want or be able to do it. I hope this project succeeds.

**Comment by xboi209** [comment]
@Alzarath Is libmpq portable? I know [StormLib](https://github.com/ladislav-zezula/StormLib) is portable and even still maintained.

**Comment by Sleaker** [comment]
@Alzarath - it's old but as I mentioned Stargus did some of it. It's more functional than this is at least.

**Comment by Alzarath** [comment]
@xboi209 I'm not a heavy programmer. If you mean if it's cross-platform, I know it's available on Linux at least, and seems to have its source code available on github. If you mean if it doesn't affect external files, I couldn't tell you.

As for maintenance, libmpq doesn't seem actively maintained. I can't tell you which one would be better for this, but libmpq does seem a little 'smaller', if that matters.

**Comment by xboi209** [comment]
@Alzarath Yes I meant cross platform. It'll be important to use a cross platform library for this since one of the strongest reasons for browser games is portability.

**Comment by andreasnrb** [comment]
Cool project but StarCraft trademark need to be defended by Blizzard or else they loose their claim to it so they will 100% guarantee nuke this repo when the news reach their legal department. To name it StarCraft, include assets is to guarantee DMCA or worse. One could ge sued etc.

**Comment by roblabla** [comment]
@andreasnrb Blizzard's legal department will only start sueing if the DMCA gets ignored. It's cheaper to send a DMCA letter than to send the lawyers after some random guy on the internet reusing your sprites.

Besides, IANAL, but If they don't follow standard procedure and send a DMCA notice first, they would risk being seen as acting in bad faith in court.

**Comment by aleksandaratanasov** [comment]
Naming of the project along with multiple art components have to be changed in order to ensure the survival of this cool project. Please do that so that we can enjoy it in the future too.

**Comment by DMGambone** [comment]
I'm not expert on this area, but my understanding is that it is completely legal to contain copies of digital content that you have paid for and own.  The same rules apply to ROMs used for MAME games.  It's illegal to have those ROMs unless you own the original arcade game.

So having these assets may be illegal to have here for free download, but having a tool that extracts them locally from a local legal copy of StarCraft should be completely legal, although voiding the EULA most likely and annoying Blizzard.

Just my 2 cents... which isn't even worth that.

**Comment by xboi209** [comment]
Well violating the EULA is indeed illegal but I think it really comes down to whether or not Blizzard will take action. For their older games, I think they will only take legal action if someone is profiting off of their work, which no one here is even thinking of doing.

**Comment by aleksandaratanasov** [comment]
@DMGambone Yes, if you own a copy of the game it should be perfectly safe to use those in a non-commercial/non-redistributable way. Some game devs/publishers however might pose additional restrictions. Dunno how it's with Blizzard on this matter. As an example for a "good behaviour" I can point at OpenTTD, which can use both its own and the art (both sprites and sound) from the original Transport Tycoon Deluxe.
